src/Vogen/Util.cs

```diff
@@ -31,7 +31,9 @@ public static string GenerateValidation(VoWorkItem workItem)
             return @$"var validation = {workItem.TypeToAugment.Identifier}.{workItem.ValidateMethod.Identifier.Value}(value);
             if (validation != Vogen.Validation.Ok)
             {{
-                throw new {workItem.ValidationExceptionFullName}(validation.ErrorMessage);
+                throw new {workItem.ValidationExceptionFullName}(validation.ErrorMessage) {{
+                    Data = validation.Data
```

Comment:
This is [failing](https://github.com/SteveDunn/Vogen/pull/375/checks#step:4:2128) with `cannot be assigned to -- it is read only`

We could either have a constructor overload, over make add an `init` accessor on the property.

Reply:
I would also argue that `Validation.Data` should not be instantiated with a new dictionary on object construction. For the common case where consumers do not provide data, this adds an allocation that is not necessary. Instead, it should be instantiated in `WithData` when the consumer adds the first data item.